workshops/json_db/README.md

```diff
@@ -0,0 +1,333 @@
+---
+
+Databases are often used in projects to persist and store data associated with certain users. In this workshop, you'll be building your very own database with the power of JavaScript!
+
+# Prerequisites
+
+Before starting this workshop, you should ideally have knowledge of:
+
+- Basic JavaScript syntax
+- Asynchronous programming (explained in workshop, but it'd be nice to have been introduced to it before)
+- Using the `npm` tool to manage dependencies
+- Basic HTTP methods
+
+# Getting Started
+
+To bootstrap and run our project, we'll need to head over to [repl.it](https://repl.it) and create a new Node project!
+
+![Create a new project](https://cloud-2m3vf7nmm.vercel.app/4new-project.png)
+
+Next, select the `Node.js` option under the `language` dropdown. You'll be redirected to a new tab in which we'll build our database! To get started, we'll need to
+
+1. Initialize an `npm` project
+2. Install the dependencies associated with it. In this case, we'll need both `express` (a web server framework) and `cors` (an express middleware).
+
+To do this, head over to the `Shell` tab and run the following:
+
+![Initialize npm and install dependencies](https://cloud-2m3vf7nmm.vercel.app/1npm-install-deps.png)
+
+You can safely ignore the warnings, they're not important to our project! Let's recap what we just did:
+
+```shell
+# Initialize an npm project in the current directory. The --yes "flag" accepts default values when prompted.
+npm init --yes
+
+# Install (shorthand: i) both the express and cors packages for our project
+npm [i/install] express cors
+```
+
+Next, let's write out a basic web server. Here's what that would look like with express:
+
+```javascript
+// "Import" the express and cors modules
+const express = require('express')
+const cors = require('cors')
+
+// Create our basic app
+const app = express()
+app.use(cors())
+
+// Run the server either on the PORT environment variable or, if that's not present, port 3000!
+const PORT = process.env.PORT || 3000
+
+// When someone sends a GET request to `/`, respond with "Hello World"!
+app.get('/', (req, res) => {
+	res.send('hello world!')
+})
+
+// Start listening on the port we specified earlier
+app.listen(PORT, () => {
+	console.log(`Listening on ${PORT}`)
+})
+```
+
+Now, let's run this! Hit the big green run button on the top of your editor to do so:
+
+![Run program](https://cloud-2m3vf7nmm.vercel.app/5run.png)
+
+Awesome! We'll head over to [hoppscotch](https://hoppscotch.io/) to finish up this basic server. Hoppscotch is an awesome website to send HTTP requests to a remote server! Let's sent a GET request the root (`/`) of our server:
+
+![Request](https://cloud-2m3vf7nmm.vercel.app/3request.png)
+
+In the screenshot above, I've copied in the URL of my REPL (in the format: https://<PROJECT_NAME>.<USERNAME>.repl.co) and set the request type to GET. Finally, hit `Send`! You should get a plain response like so:
```

Comment:
```suggestion
In the screenshot above, I've copied in the URL of my REPL (in the format: `https://<PROJECT_NAME>.<USERNAME>.repl.co`) and set the request type to GET. Finally, hit `Send`! You should get a plain response like so:
```

Reply:
without the code formatting, it was acting weird as the <USERNAME> part was not visible